Userland/Applications/PDFViewer/PDFViewerWidget.cpp

```diff
@@ -460,6 +494,21 @@ PDF::PDFErrorOr<void> PDFViewerWidget::try_open_file(StringView path, NonnullOwn
         m_sidebar_open = false;
     }
 
+    auto thumbnails_model = ThumbnailsModel::create();
+    for (u32 i = 0; i < document->get_page_count(); i++)
+        thumbnails_model->append_thumbnail(render_thumbnail_for_rendered_page(i));
```

Comment:
Synchronously rendering all thumbnails on file open doesn't seem ideal. Could we render just the ones that are visible on demand?

(Try opening a file with many pages like e.g. the PDF 1.7 spec.)

(See also #23036.)

Reply:
After implementing lazy loading thumbnails, I did some simple benchmarking on file open time, here is the result:

No thumbnails:

RTL8139 datasheet: 444 ms
PDF 1.7 spec: 5757 ms

Rendering all thumbnails at once:

RTL8139 datasheet: 2281 ms
PDF 1.7 spec: 22683 ms (oof)

Lazy loading thumbnails:

RTL8139 datasheet: 487 ms
PDF 1.7 spec: 4349 ms